core/src/main/java/org/infinispan/reactive/publisher/impl/ClusterPublisherManagerImpl.java

```diff
@@ -223,12 +227,19 @@ public void stop() {
       // This excludes the keys from the various address targets
       Map<Address, Set<K>> keyTargets = determineKeyTargets(topology, keysToInclude, localAddress, segments, ctx);
 
+      int keyTargetSize = keyTargets.size();
+
+      if (keyTargetSize == 0) {
+         flowableProcessor.onComplete();
+         return;
+      }
```

Comment:
Couldn't the context contain entries that need to be processed, while `keyTargetSize` is 0?

Reply:
Good point, I will take a look again. 👍

Reply:
This is fine for the keyPublisher as it checks the context, but the entryPublisher does not. I will add that in.

Reply:
Cool :+1: